### PR TITLE
add a design document directory and template

### PR DESCRIPTION
**Pull request by dhellmann**
This is a start at a template for design documents for larger
metalkube changes, adapted from similar documents used in OpenStack.

Signed-off-by: Doug Hellmann <dhellmann@redhat.com>

### CODE REVIEW

File: design/_template.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+# Problem description #
+
+A detailed description of the problem. What problem is this blueprint
+addressing?
+
+## Use Cases ##
+
+What use cases does this address? What impact on actors does this
+change have?  Ensure you are clear about the actors in each use case:
```
Comment (etingof) [inline design/_template.md]: nit: double space

File: design/_template.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+* Please wrap text at 79 columns.
+
+* Please do not delete any of the sections in this template.  If you have
+  nothing to say for a whole section, just write: None
+
+
+# Problem description #
+
+A detailed description of the problem. What problem is this blueprint
```
Comment (russellb) [inline design/_template.md]: s/blueprint/design/ ?

File: design/_template.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+# Implementation #
+
+## Assignee(s) ##
+
+Who is leading the writing of the code?
+
+## Work Items ##
+
+Work items or tasks -- break the feature up into the things that need to be
```
Comment (etingof) [inline design/_template.md]: Would it make sense to link the assignees to the work items?

File: design/_template.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+## Dependencies ##
+
+* Include specific references to work here or in other projects that
+  this design either depends on or is related to.
+
+* Does this feature require any new library dependencies or code
+  otherwise not included in the code? Or does it depend on a specific
+  version of library?
+
```
Comment (etingof) [inline design/_template.md]: nit: extra new line

File: design/_template.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+# Testing #
+
+Please discuss the important scenarios needed to test here, as well as
+specific edge cases we should be ensuring work correctly. For each
+scenario please specify if this requires specialized hardware, a full
+baremetal environment, or can be simulated with virtual machines.
+
```
Comment (etingof) [inline design/_template.md]: nit: extra new line

**Comment by dhellmann** [comment]
I think all of the comments are addressed by incorporating more of the KEP template into this one.

File: design/_template.md (hunk 3 above)
Comment (dhellmann) [reply design/_template.md]: Probably. They might not all be known to start, but that information could certainly be included as people pick up the tasks. We could also use github issues for tracking that, and I would expect us to link to those issues in that case.

**Comment by russellb** [comment]
Looks great!  Thanks, Doug!
